vit-servicing-station-lib/src/db/models/proposals.rs

```diff
@@ -109,46 +113,55 @@ pub struct Proposal {
     pub fund_id: i32,
     #[serde(alias = "challengeId")]
     pub challenge_id: i32,
-    #[serde(
-        alias = "proposalSolution",
-        default,
-        skip_serializing_if = "Option::is_none"
-    )]
-    pub proposal_solution: Option<String>,
-    #[serde(
-        alias = "proposalBrief",
-        default,
-        skip_serializing_if = "Option::is_none"
-    )]
-    pub proposal_brief: Option<String>,
-    #[serde(
-        alias = "proposalImportance",
-        default,
-        skip_serializing_if = "Option::is_none"
-    )]
-    pub proposal_importance: Option<String>,
-    #[serde(
-        alias = "proposalGoal",
-        default,
-        skip_serializing_if = "Option::is_none"
-    )]
-    pub proposal_goal: Option<String>,
-    #[serde(
-        alias = "proposalMetrics",
-        default,
-        skip_serializing_if = "Option::is_none"
-    )]
-    pub proposal_metrics: Option<String>,
+}
+
+#[derive(PartialEq, Eq, Debug, Clone)]
+pub enum ProposalChallengeInfo {
+    Simple(simple::ChallengeInfo),
+    CommunityChoice(community_choice::ChallengeInfo),
+}
+
+#[derive(Serialize, Deserialize)]
+struct SerdeProposalChallengeInfo {
+    #[serde(flatten, default, skip_serializing_if = "Option::is_none")]
+    simple: Option<simple::ChallengeInfo>,
+    #[serde(flatten, default, skip_serializing_if = "Option::is_none")]
+    community: Option<community_choice::ChallengeInfo>,
 }
 
 #[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
 pub struct FullProposalInfo {
     #[serde(flatten)]
     pub proposal: Proposal,
+    #[serde(flatten)]
+    pub challenge_info: ProposalChallengeInfo,
     #[serde(alias = "challengeType")]
     pub challenge_type: ChallengeType,
```

Comment:
In the ultimate perfect form of this, I'd like to eliminate the `challenge_type` member as well and synthesize the field from the `challenge_info` discriminant, but this will do for now.

Reply:
For us is ok. But for the API consumer, I don't think it is ok. Either we keep this and have the `ProposalChallengeInfo` data flattened. Or we make it nested and remove the `challenge_type`.